### ISSUE TITLE
Adding a way to exclude field of a CustomResource from restore

### ISSUE THREAD

**Issue by kaovilai** [vmware-tanzu/velero#4572]
**Describe the problem/challenge you have**
[A description of the current limitation/problem/challenge that you are experiencing.]

Some k8s custom resource contains a field which is actively monitored/managed by an [operator](https://operatorframework.io/)  controller via use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
It is possible to cause restore failures when this field exists upon restore. The field is expected to be populated automatically upon creation and therefore it would be impossible to backup without this field auto-repopulating.

A snippet from a CRD managed by a controller.
```yaml
           spec:
                version:
                  description: >-
                    This field is populated by the App Connect Operator, and
                    does not require manual input.
                  type: string
```

TL;DR: If a Custom Resource contains `spec.A` and `spec.B` at backup time, provide a way to restore just `spec.A` by include/exclude

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Some way to input specifications about what field should be removed from an object prior to restore

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by ywk253100** [comment]
Velero has a plugin architecture that allows users to add their own custom functionality to Velero restores, please refer to the [doc](https://velero.io/docs/v1.8/custom-plugins/) for more details
